input_writer.go

```diff
@@ -0,0 +1,30 @@
+package logrus
+
+import (
+	"bufio"
+	"io"
+	"runtime"
+)
+
+func (logger *Logger) InputWriter() (*io.PipeWriter) {
+	inputReader, inputWriter := io.Pipe()
+
+	go logger.inputWriterScanner(inputReader)
+	runtime.SetFinalizer(inputWriter, inputWriterFinalizer)
```

Comment:
Maybe return an `io.WriteCloser` and specify in the doc that the writer must be closed. Failing to close it is a leak, but its reasonable to expect people to close their resources. 

Since the writer should be closed, I'm not super keen on using `runtime.SetFinalizer` to make up for people writing buggy code.